apps/web/src/utils/kv/index.ts

```diff
@@ -1,166 +1,29 @@
- */
-export class KVManager {
-  private client: RedisType | null = null;
-
-  private readonly connectionArg: KvConstructorParam;
-
-  private readonly connectionTls: boolean;
-
-  constructor(connectionParam: KvConstructorParam) {
-    if (!connectionParam || (!('url' in connectionParam) && !('host' in connectionParam))) {
-      throw new Error('No URL or options provided to KVManager');
-    }
-    this.connectionArg = connectionParam;
-    this.connectionTls = connectionParam.tls ?? false;
-  }
-
-  private async getClient(): Promise<RedisType> {
-    if (!this.client) {
-      if (!this.connectionArg) {
-        throw new Error('No URL or options provided to KVManager');
-      }
-
-      if ('url' in this.connectionArg) {
-        this.client = new Redis(this.connectionArg.url, this.connectionTls ? { tls: {} } : {});
-      } else {
-        this.client = new Redis({
-          ...this.connectionArg,
-          tls: this.connectionTls ? {} : undefined,
-        });
-      }
-    }
-
-    try {
-      await this.client.ping();
-    } catch (err) {
-      if (!isDevelopment) {
-        logger.error('KV connection failed', err);
-      }
-      throw new Error(`Failed to connect to KV: ${err}`);
-    }
-
-    return this.client;
-  }
-
-  async ping() {
-    if (this.client) {
-      try {
-        return await this.client.ping();
-      } catch (err) {
-        if (!isDevelopment) {
-          logger.error('Failed to scan keys', err);
-        }
-        throw new Error(`Failed to ping: ${err}`);
-      }
-    }
-  }
-
-  async close() {
-    if (this.client) {
-      try {
-        await this.client.quit();
-        this.client = null;
-      } catch (err) {
-        if (!isDevelopment) {
-          logger.error('Failed to close client', err);
-        }
-        throw new Error(`Failed to close client: ${err}`);
-      }
-    }
-  }
-
-  async scan(cursor: number | string = '0', batchSize: number | string = 10) {
-    try {
-      const client = await this.getClient();
-      const [newCursor, elements] = batchSize
-        ? await client.scan(cursor, 'COUNT', batchSize)
-        : await client.scan(cursor);
-
-      return { cursor: newCursor, elements };
-    } catch (err) {
-      if (!isDevelopment) {
-        logger.error('Failed to scan keys', err);
-      }
-      throw new Error(`Failed to scan keys: ${err}`);
-    }
+function createKv(config: RedisConfigNodejs) {
+  if (!config.url) {
+    throw new Error('Invalid KV config');
   }
+  return new Redis({ cache: 'default', ...config });
+}
 
-  async get<T>(key: string): Promise<T | null> {
-    try {
-      const client = await this.getClient();
-      const value = await client.get(key);
-      return value ? (JSON.parse(value) as T) : null;
-    } catch (err) {
-      if (!isDevelopment) {
-        logger.error('Failed to get key', err);
+// Exports a lazy-loaded instance of KVManager with the default KV URL
```

Comment:
suggestion: Assuming that our Next.js server is a stateful Node server or similar, it's probably not worth lazy loading since we'd only be saving a really tiny amount of memory until that first request comes in.